src/django_components/util/template_tag.py

```diff
@@ -87,9 +35,106 @@ class TagSpec(NamedTuple):
     - and treated as `only=False` and `required=False` if omitted
     """
 
+    def copy(self) -> "TagSpec":
```

Comment:
We allow each component to potentially use a different template tag. E.g. `{% table ... / %}` instead of `{%  component "table" ... / %}`.

For that, we have to set the `tag` and `endtag` fields on the component's `TagSpec` instance.

But that means we also have to make copies of the original TagSpec. Hence this `copy()` function.